### ISSUE TITLE
`NoNameShadowing` false positive?

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#3990]
## Context
I am currently looking into activating more rules for the detekt code base itself and ran into an issue with the `NoNameShadowing` rule. It finds quite a few violations in _detekt_ and I don't understand why that is the case.
 

For example, this code is flagged: 

```kotlin
fun doStuff() = listOf("")
    .let { list ->
        list.map { it + "x" }
    }
```

Replacing `list.map { it + "x" }` with `list.map { a -> a + "x" }` solves the problem. Is that behavior intended?

**Comment by schalkms** [comment]
Without having the full code context, I would say that this isn't intended and therefore a false-positive.

**Comment by t-kameyama** [comment]
Reproduced it. I will fix this issue.
